### PR TITLE
Multiple positions

### LINKED ISSUE #5784: MultiApp Input-file/positions enhancement
@mellis13 needs a small enhancement to MultiApps.  Currently if you use `positions_from_file` you can only use one input file.

Instead... we should allow one input file per `positions_with_file`.

### PR DESCRIPTION
**Pull request by friedmud**
Give MultiApps the ability to specify multiple input files with multiple position files.

Also... various cleanups and error checking tests...

closes #5784

### CODE REVIEW

**Comment by friedmud** [comment]
A fix for Yak is here https://hpcgitlab.inl.gov/idaholab/yak/merge_requests/336

File: framework/src/multiapps/MultiApp.C [hunk 1]
```diff
@@ -173,35 +171,67 @@ MultiApp::initialSetup()
...
+    std::vector<FileName> input_files = getParam<std::vector<FileName> >("input_files");
+
+    if (input_files.size() != 1 && positions_files.size() != input_files.size())
+      mooseError("Number of input_files for MultiApp " << name() << " must either be only one or match the number of positions_file files");
+
+    // Clear out the _input_files because we're going to rebuild it
+    if (input_files.size() != 1)
+      _input_files.clear();
+
+    for (unsigned int p_file_it = 0; p_file_it < positions_files.size(); p_file_it++)
```
Comment (permcody) [inline framework/src/multiapps/MultiApp.C]: interesting variable name for a counter (not an iterator)

Comment (friedmud) [inline framework/src/multiapps/MultiApp.C]: Well - I needed a counter at the time and didn't want to name it `k`.  I could change it now to an actual iterator.  The current code doesn't rely on it being a counter.

Comment (permcody) [inline framework/src/multiapps/MultiApp.C]: That's not necessary. I just noticed it, that's all.

**Comment by friedmud** [comment]
This is ready to go as far as I know.  After merging someone will need to click merge on that Yak MR.

**Comment by YaqiWang** [comment]
@friedmud Your MR broke few other tests due to changing from std::string to FileName.

**Comment by permcody** [comment]
Well more than that. The PR in Yak will require a MOOSE submodule update
(to the yet to be determined hash in MOOSE) in order to flow through the
system.

**Comment by friedmud** [comment]
@YaqiWang Which MR?  This one or the one I put into Yak?  Yeah.. the two things had to go together... this PR and the MR I proposed to Yak.

@permcody How did all of the merging go?  Is there anything I can do in the future to do this better?

**Comment by permcody** [comment]
You did as much as you could do without having the first one merged into
MOOSE. I merged it into MOOSE then updated the submodule in Yak and pushed
your commit and mine as a new MR which Yaqi merged. You can't get the
submodule hash without the first patch merged.

**Pull request merged**